### ISSUE TITLE
[IDEA] RevealWidget attribute for scrolling to the dropdown 

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#7498]
**Is your feature request related to a problem? Please describe.**
It is annoying to click a dropdown only to have to scroll downwards in order to see the dropdown contents.

**Describe the solution you'd like**
 A RevealWidget attribute, e.g `scrollto=yes` that makes the dropdown scroll so its top edge to aligns with viewport top.  Or, more fine tuned; `scrollto=top` to behave as just described (used for dropdowns that are longer than viewport) and `scrollto=bottom` to align dropdown bottom edge with viewport bottom (used for dropdowns that are shorter than viewport so you still keep more of the parent tiddler in view above the revealwidget).

Maybe additional attribute `scrollback=yes` used as a complimentary attribute to navigate back to the tiddler when revealwidget is closed i.e align tiddler with viewport top again.

**Comment by pmario** [comment]
I think it's possible to implement. But the question is, do you want it to be enabled at tiddlywiki.com?

If yes, I think that's a bit hard to implement in a sensible way, that will work for every user and every screen-size. I personally do not like apps that try to be smart and do aggressive navigations automatically. We need to take care, that users do not loose the focus. 

On my mobile phone it does scroll the input text area into the middle of the readable screen. Just a little bit, so it's not disruptive. 

On a PC, I actually can not think about a usecase, where I did have a problem, with using the mouse wheel to scroll down. ... Can you post a screenshot or a short info, how to replicate the problem?

**Comment by twMat** [comment]
No, I had not really considered it for tiddlywiki.com, I need it for myself. 

Real use case, well one case we all are affected by is the editview `type` dropdown. But that's mostly to illustrate the point as it would require overwrites etc if one would want that particular dropdown to do this. Otherwise, use cases are pretty much what the OP says; If the dropdown is long. Or when the a button is close to the viewport bottom edge even if the dropdown is not long. Not sure what kind of use cases I can give other than those general descriptions :)
